tls/s2n_handshake_io.c

```diff
@@ -1040,27 +1015,41 @@ int s2n_negotiate(struct s2n_connection *conn, s2n_blocked_status * blocked)
         this = 'C';
     }
 
-    while (ACTIVE_STATE(conn).writer != 'B') {
-        /* Flush any pending I/O or alert messages */
-        GUARD(s2n_flush(conn, blocked));
+    /* If we were blocked reading a record, call the handler right away. Don't try to read more data. */
+    if (conn->handshake.blocked_on_read) {
```

Comment:
As we are merging SESSION_RESUMPTION state to CLIENT_HELLO, this whole block somehow overlaps with the logic in `handshake_read_io` (except delete session). Can we reuse the existing code?